File: Synapse.js
const LEARNING_RATE = 1/1;

class Synapse {
  constructor(parentNeuron) {
    this.weight = (Math.random()) * (Math.random() < 0.5 ? -1 : 1);
    this.parentNeuron = parentNeuron;
    this.gradient = 0;
  }

  getWeightedValue() {
    if (this.parentNeuron.synapses.length > 0)
      return this.weight * this.parentNeuron.getActivationSigmoid();
    else {
      return this.weight * this.parentNeuron.activation;
    }
  }

  applyGradient() {
    this.weight -= this.gradient * LEARNING_RATE;
    this.gradient = 0;
  }
}

module.exports = Synapse;

File: python/network/Synapse.py
from random import random

LEARNING_RATE = 1/10


class Synapse:
    def __init__(self, parent_neuron):
        self.weight = (random() * (-1 if random() < 0.5 else 1))
        self.parent_neuron = parent_neuron
        self.gradient = 0

    def get_weighted_value(self):
        return self.weight * self.parent_neuron.get_activation_sigmoid()

    def applyGradient(self):
        self.weight = self.weight - (self.gradient * LEARNING_RATE)
        self.gradient = 0
